extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/CdiAwareJob.java

```diff
@@ -39,16 +38,9 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
 
     @Override
     public void interrupt() throws UnableToInterruptJobException {
-        Instance.Handle<? extends Job> handle = jobInstance.getHandle();
         // delegate if possible; throw an exception in other cases
         if (InterruptableJob.class.isAssignableFrom(handle.getBean().getBeanClass())) {
-            try {
-                ((InterruptableJob) handle.get()).interrupt();
-            } finally {
-                if (handle.getBean().getScope().equals(Dependent.class)) {
-                    handle.destroy();
-                }
-            }
+            ((InterruptableJob) handle.get()).interrupt();
```

Comment:
This may call `interrupt()` on already destroyed instance. Though, not sure if this should be handled somehow...

Reply:
That is something the bean (the job) should take into consideration. I don't think we can prevent it because the way interrupt and execute interact depend on the the job implementation anyway.
ATM we are delaying the destruction as far as we can - after the `execute()` is done.

Reply:
> That is something the bean (the job) should take into consideration. I don't think we can prevent it because the way interrupt and execute interact depend on the the job implementation anyway.

Job implementation can control only internals of `CustomInterruptableJob.execute(context)`, and `CustomInterruptableJob.interrupt()`. It cannot control work of the CdiAwareJob wrapper. For example,

1. CustomInterruptableJob.execute(context) is finished and goes out of control of the CustomInterruptableJob.
2. Instance.Handle.destroy() is finished.
3. Accidentally, CdiAwareJob.interrupt() is invoked and fails with "Instance already destroyed". The control is not even passed to the CustomInterruptableJob.interrupt().

So this is not a matter of custom `interrupt` and `execute` interaction, it is a matter of CdiAwareJob's `interrupt` and `execute` interaction.

Reply:
Hm, Quartz impl only calls `interrupt()` if it finds a running job but it doesn't seem to care about race conditions which is what you imply IIUIC.

In that case, for a dependent bean that has been destroyed prior to interrupt we could:
* Not invoke interrupt at all
* Keep contextual reference and invoke it anyway even though bean's `@PreDestroy` was already triggered

I assume the latter would be more user friendly as there can be logging or general logic not related to this bean that you want to happen when interruption is invoked.

Reply:
Actually I was wrong, this is more complex because of how re-firing jobs work.
A job throwing a proper exception can trigger a re-fire ([see here](https://github.com/quarkusio/quarkus/blob/main/extensions/quartz/deployment/src/test/java/io/quarkus/quartz/test/DependentBeanJobTest.java#L187-L215)) which *will use the same job instance* and we want a new bean for that.